### PR TITLE
Create part 4 of the OTA blog series

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by akbennett** [comment]
Lgtm. Great job producing something so concise

**Comment by doanac** [comment]
cool - i'll wait for feedback from our company editor before I merge .

File: src/content/blog/ota-part-4.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
++++
+title = "Securing OTA Community Edition"
+date = "2018-07-12"
+tags = ["ota", "open source"]
+categories = ["FOTA"]
+banner = "img/banners/ota.png"
+author = "Andy Doan"
++++
+
+[Part three]({{< ref "ota-part-3.md" >}}) of this blog series showed how to
```
Comment (mbolivar) [inline src/content/blog/ota-part-4.md]: This is the final one, right? Would be good to put a big list of all of them here then so people don't have to chase as many pointers to get to part 1 (and extra nice to have for going back to the previous ones to do the same).

File: src/content/blog/ota-part-4.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    - `tuf-reposerver.<ingress_dns_name>`*(now located at `api.<ingress_dns_name>/repo`)*
+
+### Secure the Reverse Proxy
+
+OTA Community Edition has no authentication support built into it, so
+securing it requires making changes to the reverse proxy we created in
+the previous blog. Nginx includes the
+[ngx_http_auth_request_module](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html)
+for securing calls to reverse proxy servers. The approach taken in this
+blog is to create a authentication service to handle `auth_requests`.
```
Comment (mbolivar) [inline src/content/blog/ota-part-4.md]: "an" authentication

File: src/content/blog/ota-part-4.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+The ota-blog-part4 GitHub
+[branch](https://github.com/OpenSourceFoundries/ota-community-edition/tree/ota-blog-part4)
+includes a new service called `oauthful` (oauth + awful). Its
```
Comment (mbolivar) [inline src/content/blog/ota-part-4.md]: It's

File: src/content/blog/ota-part-4.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  # https://github.com/OpenSourceFoundries/ota-community-edition/compare/ota-blog-part3...ota-blog-part4
+
+  # Deploy changes with:
+  ./contrib/gke/make start-services
+~~~
+
+### Verify the APIs Are Secure
+~~~
+  # Set the URL below to match your domain, and verify you get HTTP 401 errors:
+  curl -v http://api.example.com/treehub/api/v3/config
```
Comment (mbolivar) [inline src/content/blog/ota-part-4.md]: Just confirming -- you meant http and not https, right?

File: src/content/blog/ota-part-4.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  curl -H "Authorization: Bearer BadT0ken5" http://api.example.com/repo/api/v1/user_repo/root.json
+~~~
+
+The web interface will also now be secured with HTTP basic authentication.
+The username is ignored, so you can enter anything into that value. The
+password is `BadT0ken5`.
+
+### Update credentials.zip
+
+As a result of the URL changes, we'll need to update our credentials.zip that
```
Comment (mbolivar) [inline src/content/blog/ota-part-4.md]: Not sure what "our" credential.zip is - foundries.io's?

File: src/content/blog/ota-part-4.md (hunk 4 above)
Comment (doanac) [reply src/content/blog/ota-part-4.md]: Correct - in the "Next Steps" I note that has to be dealt with.

**Pull request merged**